.github/workflows/typechecks.yml

```diff
@@ -0,0 +1,52 @@
+name: Tests, Linter & Typecheck
+
+on:  
+  push:
+    branches:
+      - main
+  pull_request: 
+    types: 
+      - opened
+
+jobs:  
+  lint:
+    name: 'Linting Markdown: ${{ matrix.os }} (node@${{ matrix.node_version }})'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node_version: [16.7]
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Node.js 16.7
```

Comment:
```suggestion
      - name: Setup Node.js ${{ matrix.node_version }}
```

Reply:
Currently dont have a matrix setup, or know what Node version are supported. This is just the Wrangler supported Node version.

Reply:
I can add something like 
`strategy:
      matrix:
        node-version: [16.x, 17.x, 18.x]`